### PR TITLE
OPENEUROPA-1259: Implement "Dialogs" pattern

### PR DESCRIPTION
**Pull request by imanoleguskiza**

### CODE REVIEW

File: oe_theme.theme [hunk 1]
```diff
@@ -602,3 +613,22 @@ function oe_theme_preprocess_paragraph__oe_content_row__variant_inpage(&$variabl
...
+  if ($context->isOfType('preview')) {
+    // Loop through each already existing suggestions.
+    array_walk(
+      $suggestions,
+      function (&$suggestion) {
+        $suggestion .= '__preview';
+      }
+    );
+
+    $suggestions[] = $variables['theme_hook_original'] . '__preview';
```
Comment (ademarco) [inline oe_theme.theme]: @drupol as discussed I've added this upstream in UI Patterns (latest dev-master) so you can remove this hook implementation.

Comment (drupol) [reply oe_theme.theme]: Done bruh.

File: js/dialogs.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/**
+ * @file
+ * ECL dialog behavior.
+ */
+(function (ECL, Drupal) {
+  Drupal.behaviors.eclDialogs = {
+    attach: function () {
+      Array.from(document.querySelectorAll('[data-ecl-dialog]')).map(function(el) {
```
[the remark this answers is not in the document]
Comment (drupol) [reply js/dialogs.js]: Fixed.

File: js/dialogs.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        // Getting attribute value of elements having a data-ecl-dialog attribute.
```
[the remark this answers is not in the document]
Comment (drupol) [reply js/dialogs.js]: Fixed.

File: js/dialogs.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        return el.getAttribute('data-ecl-dialog');
+      }).filter(function onlyUnique(value, index, self) {
+        // Remove duplicates.
```
[the remark this answers is not in the document]
Comment (drupol) [reply js/dialogs.js]: Fixed.

File: oe_theme.theme [hunk 5]
```diff
@@ -15,7 +15,7 @@ use Drupal\Component\Utility\Random;
 /**
  * Implements hook_preprocess_breadcrumb().
  */
-function oe_theme_preprocess_breadcrumb(&$variables) {
+function oe_theme_preprocess_breadcrumb(array &$variables) {
```
[the remark this answers is not in the document]
Comment (drupol) [reply oe_theme.theme]: Fixed.

File: templates/patterns/dialog/dialog.ui_patterns.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      preview: "true"
+    dialog_title:
+      type: "object"
+      label: "Dialog title"
+      description: "Title of the dialog"
+      preview:
+        value: "Dialog"
+        id: "dialog-title"
+        classes: "custom-class"
+    dialog_description:
```
[the remark this answers is not in the document]
Comment (drupol) [reply templates/patterns/dialog/dialog.ui_patterns.yml]: Fixed.

File: templates/patterns/dialog/pattern-dialog--variant-normal--preview.html.twig [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{#
+/**
+ * @file
+ * Dialog component.
+ */
+#}
+<a title="Click to test the modal" href="#dialog-id" class="ecl-link" data-ecl-dialog="ecl-dialog">Open dialog</a>
+{% embed '@ecl/ec-component-dialog' with {
```
[the remark this answers is not in the document]
Comment (drupol) [reply templates/patterns/dialog/pattern-dialog--variant-normal--preview.html.twig]: Fixed.

File: composer.json [hunk 8]
```diff
@@ -7,7 +7,7 @@
     "prefer-stable": true,
     "require": {
         "drupal/core": "^8.6",
-        "drupal/ui_patterns": "dev-1.x",
+        "drupal/ui_patterns": "dev-1.x#d0a83b88cebaf246c05626b4c5b9eb5a015b3867",
```
[the remark this answers is not in the document]
Comment (drupol) [reply composer.json]: Fixed.

File: patches/@ecl/generic-component-language-list+1.1.0.patch [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+-    <div id="ecl-overlay-language-list" class="ecl-dialog__overlay{% if _system != 'eu' %} ecl-dialog__overlay--blue{% endif %}" aria-hidden="true"></div>
+     {% include '@ecl/' ~ _system ~ '-component-lang-select-site' with {
+-      'extra_attributes': [{name: 'data-ecl-dialog', value: 'ecl-dialog'}, {name: 'id', value: 'ecl-lang-select-sites__overlay'}],
++      'extra_attributes': [{name: 'data-ecl-dialog', value: _dialog_id}, {name: 'data-ecl-dialog-overlay', value:'ecl-overlay-language-list'}],
+       'extra_classes': 'ecl-link',
+       'link': _link,
+       'language': _selected_language.label,
+       'code': _selected_language.lang
+     } only %}
++    {# Custom Dialog-overlay for the 'overlay' variant #}
```
[the remark this answers is not in the document]
Comment (drupol) [reply patches/@ecl/generic-component-language-list+1.1.0.patch]: ... ?

File: oe_theme.theme [hunk 10]
```diff
@@ -391,7 +391,17 @@ function oe_theme_preprocess_links__oe_multilingual_content_language_block(&$var
 /**
  * Implements hook_preprocess_HOOK().
  */
-function oe_theme_preprocess_pattern_accordion(&$variables) {
+function oe_theme_preprocess_pattern_dialog(array &$variables) {
+  // If no identifier is provided generate a random, unique one.
```
[the remark this answers is not in the document]
Comment (drupol) [reply oe_theme.theme]: Fixed.

File: templates/patterns/dialog/dialog.ui_patterns.yml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    wide:
+      label: "Wide"
+      description: "Contrary to the normal dialog, when shown this variant of the dialog will always fill the entire screen no matter the user’s screen size."
+  fields:
+    dialog_aria_hidden:
+      type: "string"
+      label: "aria-hidden"
+      description: "'aria-hidden' of the dialog."
+      preview: "true"
+    dialog_title:
```
[the remark this answers is not in the document]
Comment (drupol) [reply templates/patterns/dialog/dialog.ui_patterns.yml]: Fixed.

File: templates/patterns/dialog/dialog.ui_patterns.yml [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+  description: "A dialog is an application window that is designed to interrupt the current processing of an application in order to prompt the user to enter information or require a response."
+  variants:
+    normal:
+      label: "Normal"
+      description: " Will display an horizontally and vertically centered container with the text."
+    wide:
+      label: "Wide"
+      description: "Contrary to the normal dialog, when shown this variant of the dialog will always fill the entire screen no matter the user’s screen size."
+  fields:
+    dialog_aria_hidden:
```
[the remark this answers is not in the document]
Comment (drupol) [reply templates/patterns/dialog/dialog.ui_patterns.yml]: Fixed, I used 'hidden'.

File: templates/patterns/dialog/dialog.ui_patterns.yml [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+      type: "string"
```
[the remark this answers is not in the document]
Comment (drupol) [reply templates/patterns/dialog/dialog.ui_patterns.yml]: Fixed.

**Pull request merged**